### PR TITLE
Allow global omit_containers["ddev-router"], add project config to bind_all_interfaces, fixes #3028, fixes #3184

### LINKED ISSUE #3028: Allow disabling ddev-router and relying on other routing for ddev-webserver http port
**Is your feature request related to a problem? Please describe.**

Setting up ddev in a Gitpod workspace requires adding settings to override ddev expectations for a local machine setup.

**Describe the solution you'd like**

I would like ddev to have a new setting/flag that "understands" cloud environments, would allow the following:
* No longer need to specify the dynamic fqdns.
* Ddev will start and display: `Your project can be reached at`, with the correct URL that is based on the cloud environment, and without assuming a `ddev.site` url.
* Ddev would not try using mkcert.
* Ddev will start without displaying these warnings in Gitpod:
```
Pushed mkcert rootca certs to ddev-global-cache/mkcert 
Running   Command=ip address show dev docker0
The hostname 8025-orange-rooster-ckv1b5zu.ws-us08.gitpod.io is not currently resolvable, trying to add it to the hosts file 
ddev needs to add an entry to your hostfile.
It will require administrative privileges via the sudo command, so you may be required
to enter your password for sudo. ddev is about to issue the command: 
    sudo /home/linuxbrew/.linuxbrew/Cellar/ddev/1.17.2/bin/ddev hostname 8025-orange-rooster-ckv1b5zu.ws-us08.gitpod.io 127.0.0.1 
Please enter your password if prompted. 
Running   Command=sudo /home/linuxbrew/.linuxbrew/Cellar/ddev/1.17.2/bin/ddev hostname 8025-orange-rooster-ckv1b5zu.ws-us08.gitpod.io 127.0.0.1
Failed to execute sudo command, you will need to manually execute '/home/linuxbrew/.linuxbrew/Cellar/ddev/1.17.2/bin/ddev hostname 8025-orange-rooster-ckv1b5zu.ws-us08.gitpod.io 127.0.0.1' with administrative privileges 
The hostname 8036-orange-rooster-ckv1b5zu.ws-us08.gitpod.io is not currently resolvable, trying to add it to the hosts file 
ddev needs to add an entry to your hostfile.
It will require administrative privileges via the sudo command, so you may be required
to enter your password for sudo. ddev is about to issue the command: 
    sudo /home/linuxbrew/.linuxbrew/Cellar/ddev/1.17.2/bin/ddev hostname 8036-orange-rooster-ckv1b5zu.ws-us08.gitpod.io 127.0.0.1 
Please enter your password if prompted. 
Running   Command=sudo /home/linuxbrew/.linuxbrew/Cellar/ddev/1.17.2/bin/ddev hostname 8036-orange-rooster-ckv1b5zu.ws-us08.gitpod.io 127.0.0.1
Failed to execute sudo command, you will need to manually execute '/home/linuxbrew/.linuxbrew/Cellar/ddev/1.17.2/bin/ddev hostname 8036-orange-rooster-ckv1b5zu.ws-us08.gitpod.io 127.0.0.1' with administrative privileges 
The hostname 8080-orange-rooster-ckv1b5zu.ws-us08.gitpod.io is not currently resolvable, trying to add it to the hosts file 
ddev needs to add an entry to your hostfile.
It will require administrative privileges via the sudo command, so you may be required
to enter your password for sudo. ddev is about to issue the command: 
    sudo /home/linuxbrew/.linuxbrew/Cellar/ddev/1.17.2/bin/ddev hostname 8080-orange-rooster-ckv1b5zu.ws-us08.gitpod.io 127.0.0.1 
Please enter your password if prompted. 
Running   Command=sudo /home/linuxbrew/.linuxbrew/Cellar/ddev/1.17.2/bin/ddev hostname 8080-orange-rooster-ckv1b5zu.ws-us08.gitpod.io 127.0.0.1
Failed to execute sudo command, you will need to manually execute '/home/linuxbrew/.linuxbrew/Cellar/ddev/1.17.2/bin/ddev hostname 8080-orange-rooster-ckv1b5zu.ws-us08.gitpod.io 127.0.0.1' with administrative privileges 
```

**Describe alternatives you've considered**

**Additional context**

This is how we add the current dynamic fqdns in ddev-gitpod:
https://github.com/shaal/ddev-gitpod/blob/main/.ddev/gitpod-setup-ddev.sh

### LINKED ISSUE #3184: Docs section for CMS-specific best practices and commonly-asked questions
Add a section in the docs to deal with commonly-asked CMS-specific questions (and best practices). Link to better docs if they are available.

### PR DESCRIPTION
**Pull request by rfay**
## The Problem/Issue/Bug:

For some very special situations, where there is already a reverse proxy provided (like gitpod) it makes sense to turn off ddev-router and let the other reverse proxy do the job.

Also, people sometimes like to have ports exposed on all network interfaces, and `bind_all_interfaces: true` allows this. In most contexts it's a security problem, but maybe not if you want to share on your local network with colleagues, etc.

* #3028 
* #3184 

## How this PR Solves The Problem:

Allow global `omit_containers["ddev-router"]`
Allow project config `bind_all_interfaces: true`
Allow `host_phpmyadmin_port: 8036`
Allow `host_mailhog_port: 8025`

## Manual Testing Instructions:

`ddev config global --omit-containers=ddev-router,ddev-ssh-agent,dba` and see how it works out.

Just use the gitpod link here and experiment on gitpod.

On gitpod, you'll want to do these things:

```yaml
bind_on_all_interfaces: true
host_webserver_port: 8080
# Will ignore the direct-bind https port, which will land on 2222
host_https_port: 2222
# phpmyadmin won't be active unless the dba container is enabled
host_phpmyadmin_port: 8036
# I don't know that this would be useful on gitpod, but allows local db clients to run.
host_db_port: 3306
# Probably want to ignore these two ports
host_mailhog_port:  8025
```

```
ddev poweroff
ddev config global --omit-containers=ddev-router,ddev-ssh-agent,dba
ddev config --project-type=drupal9 --docroot=web --create-docroot
ddev composer create drupal/recommended-project -y
```

## Automated Testing Overview:

This is a very unusual usage, and I didn't add any tests.

## Release/Deployment notes:

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/3191"></a>

### CODE REVIEW

**Comment by rfay** [comment]
Binaries for testing this are at https://github.com/drud/ddev/actions/runs/1152600782

File: pkg/globalconfig/global_config.go [hunk 1]
```diff
@@ -159,9 +158,10 @@ func WriteGlobalConfig(config GlobalConfig) error {
 	// Append current image information
 	instructions := `
 # You can turn off usage of the dba (phpmyadmin) container and/or
-# ddev-ssh-agent containers with
-# omit_containers["dba", "ddev-ssh-agent"]
-# and you can opt in or out of sending instrumentation the ddev developers with
+# ddev-ssh-agent and ddev-router containers with
+# omit_containers["dba", "ddev-ssh-agent", "ddev-router"]
+
+# You can opt in or out of sending instrumentation the ddev developers with
```
Comment (shaal) [inline pkg/globalconfig/global_config.go:164]:
```suggestion
# You can opt in or out of sending instrumentation to the ddev developers with
```

**Review by shaal — APPROVED** [review]
Thank you @rfay for making this happen!!

I tested this PR using Gitpod button on the PR, and following the instructions.
Work exactly as expected.

I also test this PR using DrupalPod. I downloaded ddev binaries from this PR, and it's a dream come true, running `ddev start` in Gitpod runs without any of the previous 'warnings'.

I'm also looking forward for Gitpod hiding all the ports that are not viewable.
https://github.com/gitpod-io/gitpod/issues/5303

Please let me know if there's any additional tests I can run.
I cannot wait for this to be released! @rfay Thank you, Thank you!

**Review by shaal — APPROVED** [review]
Works great!
Tested on DrupalPod, with https://github.com/shaal/DrupalPod/pull/34 

Thank you @rfay

**Pull request merged**